plugins/inputs/prometheus_scraper/metrics_type_handler.go

```diff
@@ -153,19 +139,24 @@ func isInternalMetric(metricName string) bool {
 	return false
 }
 
-// Decorate the Metrics with Metric Types
+// Decorate the Metrics with Metric Types.
+// Filter out Summary, Histogram and untyped Metrics and adding logging.
 func (mth *metricsTypeHandler) Handle(pmb PrometheusMetricBatch) (result PrometheusMetricBatch) {
-	// Filter out Summary, Histogram and untyped Metrics and adding logging
-	jobName, instanceId, err := GetScrapeTargetInfo(pmb)
-	if err != nil {
-		log.Printf("E! Failed to get Job Name and Instance ID from Prometheus metrics. \n")
-		return result
+	if len(pmb) == 0 {
+		log.Printf("D! Skip empty batch")
+		return nil
+	}
+
+	// All metrics in a batch are from same scrape target, we just need first one
+	jobName, instanceId := pmb[0].jobBeforeRelabel, pmb[0].instanceBeforeRelabel
```

Comment:
Do we need to check all metrics in the batch like GetScrapeTargetInfo does?

Reply:
Yeah, we can do that, I can reprogram `GetScrapeTargetInfo` I should have deleted that func because it is only used here.

Reply:
Solved in https://github.com/aws/amazon-cloudwatch-agent/pull/194/commits/26bc57b52c3858b96a4654d83b4f95831bcd2ef7